scipy/stats/tests/test_resampling.py

```diff
@@ -638,6 +665,35 @@ def statistic(x, axis):
         assert_allclose(res.statistic, expected.statistic)
         assert_allclose(res.pvalue, expected.pvalue, atol=self.atol)
 
+    def test_vectorized(self):
```

Comment:
All tests are very similar. Did you try to parametrize or even have a `if` for options?

Reply:
ok because I would have thought that just having `fun, options, rvs` as param should be enough.

If you still prefer to keep 3 tests, fine. Just maybe only define once the 2 functions `statistic`. Also `options` does not need to be a function.

Reply:
It would not, if you'd permit me to use the global random state. In fact, if I could do that, I'd be happy to combine all the tests. All of the difficulty in combining them is because I have to worry about passing Generators.

Reply:
Alternatively, I can combine the tests if we don't check that the results match - only that the assertions don't fail.

Reply:
Not sure about the generator part. But fine with me not to check the results as we have other tests for all that.